### PR TITLE
pass messages not from child to handle_other

### PR DESCRIPTION
**Pull request by andpodob**
Fix the issue of handling `:DOWN` messages from processes that are not children of pipeline. This PR makes `develop` to work properly when crash groups are not used. Crash groups are yet to be debugged.

### CODE REVIEW

File: lib/membrane/core/parent/child_life_controller.ex [hunk 1]
```diff
@@ -146,44 +146,43 @@ defmodule Membrane.Core.Parent.ChildLifeController do
...
-          handle: {:ok, state} <- LifecycleController.maybe_finish_playback_transition(state) do
-      {{:ok, :child}, state}
+  @spec handle_child_death(child_pid :: pid(), reason :: atom(), state :: Parent.state_t()) ::
+          {:ok | {:error, :not_child}, Parent.state_t()}
+  def handle_child_death(pid, :normal, state) do
+    with {:ok, child_name} <- child_by_pid(pid, state),
+         state = Bunch.Access.delete_in(state, [:children, child_name]),
+         state = remove_child_links(child_name, state),
+         {:ok, state} <- LifecycleController.maybe_finish_playback_transition(state) do
+      {:ok, state}
```
Comment (mat-hek) [inline lib/membrane/core/parent/child_life_controller.ex:156]:
```suggestion
    with {:ok, child_name} <- child_by_pid(pid, state) do
       state = Bunch.Access.delete_in(state, [:children, child_name])
       state = remove_child_links(child_name, state)
       LifecycleController.maybe_finish_playback_transition(state)
```

**Review by mat-hek — CHANGES_REQUESTED** [review]

File: lib/membrane/core/parent/child_life_controller.ex (hunk 1 above)
Comment (andpodob) [reply lib/membrane/core/parent/child_life_controller.ex:156]: `maybe_finish_playback_transition(state)` is having specs for return value like this: `{:ok | {:error, any}, Parent.state_t()}`, is that safe to just return result of `LifecycleController.maybe_finish_playback_transition(state)`?

Comment (mat-hek) [reply lib/membrane/core/parent/child_life_controller.ex:156]: Specs (and possibly error handling) for `maybe_finish_playback_transition` should be fixed, but the code I suggested works exactly the same way as yours, unless I'm missing something

File: lib/membrane/core/parent/child_life_controller.ex [hunk 2]
```diff
@@ -144,23 +144,23 @@ defmodule Membrane.Core.Parent.ChildLifeController do
...
-         {:ok, state} <- LifecycleController.maybe_finish_playback_transition(state) do
-      {:ok, state}
+         state = remove_child_links(child_name, state) do
+      LifecycleController.maybe_finish_playback_transition(state)
     else
-      {:error, :not_child} -> {{:error, :not_child}, state}
-      error -> error
+      {:error, :not_child} ->
+        raise Membrane.PipelineError,
+              "Tried to handle child death that wasn't a child of that pipeline."
```
Comment (mat-hek) [inline lib/membrane/core/parent/child_life_controller.ex:159]: fix english :P

File: lib/membrane/core/parent/child_life_controller.ex [hunk 3]
```diff
@@ -144,23 +144,23 @@ defmodule Membrane.Core.Parent.ChildLifeController do
...
   """
   @spec handle_child_death(child_pid :: pid(), reason :: atom(), state :: Parent.state_t()) ::
           {:ok | {:error, :not_child}, Parent.state_t()}
   def handle_child_death(pid, :normal, state) do
     with {:ok, child_name} <- child_by_pid(pid, state),
          state = Bunch.Access.delete_in(state, [:children, child_name]),
-         state = remove_child_links(child_name, state),
-         {:ok, state} <- LifecycleController.maybe_finish_playback_transition(state) do
-      {:ok, state}
+         state = remove_child_links(child_name, state) do
```
Comment (mat-hek) [inline lib/membrane/core/parent/child_life_controller.ex]: assignments can be moved out of with

File: lib/membrane/core/parent/child_life_controller.ex [hunk 4]
```diff
@@ -169,7 +169,9 @@ defmodule Membrane.Core.Parent.ChildLifeController do
 
       {:ok, state}
     else
-      {:error, :not_member} = error -> {error, state}
+      {:error, :not_member} ->
+        raise Membrane.PipelineError,
+              "Child  that was not a member of a group killed with :membrane_crash_group_kill."
```
Comment (mat-hek) [inline lib/membrane/core/parent/child_life_controller.ex:174]:
```suggestion
              "Child that was not a member of a crash group killed with :membrane_crash_group_kill."
```

**Review by mat-hek — CHANGES_REQUESTED** [review]

**Review by mat-hek — APPROVED** [review]

**Pull request merged**